File: Samples/Unmanaged/PingPongWrapper/Events.cpp
#include <memory>

#include "Events.h"

Events::ConfigEvent::ConfigEvent(Microsoft::PSharp::MachineId^ mid)
	: Microsoft::PSharp::Event()
{
	this->id = mid;
}

Events::ConfigEvent::~ConfigEvent() { }

Events::MessageEvent::MessageEvent()
	: Microsoft::PSharp::Event()
{

}

Events::MessageEvent::~MessageEvent() { }

File: Samples/PSharpAsLibrary/PingPong/Events.cs
using Microsoft.PSharp;

namespace PingPong
{
    #region Events

    internal class Config : Event
    {
        public MachineId Id;

        public Config(MachineId id)
            : base()
        {
            this.Id = id;
        }
    }

    internal class Unit : Event { }
    internal class Ping : Event { }
    internal class Pong : Event { }

    #endregion
}
